### DISCUSSION TITLE
Can headless mode work without activating login session in advanced

### DISCUSSION THREAD

**Discussion by liyimeng** [rustdesk/rustdesk#5979] [General]
### Bug Description

If I try to connect to my machine in cloud, which I have setup rustdesk in headless mode. It fails. To workaround this issue, I have to login my machine and activate an login session in advanced, e.g. using lightdm. 

This is not the case if I use plain xrdp, in which, I was prompted for enter username and password, it then start a session for me. 

I check into the code and find rustdesk rely on systemd tool to find out session information, which failed since there is no active session there. I am curious about two things:

- Why an active session is needed in advanced, why rustdesk could not behave like xrdp?
- Is it possible to develop a systemd less solution?  how much work to to be done there? because of lightweight of rustdesk, I am planning to use it in a resource constraint environment. Those environments normally run without systemd, which is regarded as a bloatware for many. :) 

Thanks a lot in advanced! 

### How to Reproduce

- setup a headless rustdesk
- connect to it, failed with "Unsupported display server type"

### Expected Behavior

work as xrdp

### Operating system(s) on local side and remote side

osx --> linux

### RustDesk Version(s) on local side and remote side

1.2.3 --> 1.2.3

### Screenshots

something like this 
![](https://user-images.githubusercontent.com/37828835/274111586-8c0917b0-de11-45c5-a7f9-1cbd28d0ac0d.png)

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/wiki/Headless-Linux-Support

**Reply by liyimeng** [reply]
It is where I started from :)     I could not find an answer.   Anymore hints :). @rustdesk

**Reply by ihtarlik** [reply]
I, too, am having this issue. I have Rustdesk relay server on a VPS running Ubuntu 24.04.2 LTS. I also installed the Rustdesk client on the VPS and on my Android, and both use the relay server info. I followed the headless instructions listed above (installing requisite packages, enabled headless option, getting Rustdesk ID, and setting password). If I have signed into the system using XRDP at least once, then the Android connects to the VPS without issue. However, on a fresh boot, I get "Connection Error: Remote desktop is offline."

This is occurring with both clients running version 1.3.8.
